### PR TITLE
Adding docs to clarify environments vs environments + additional tests

### PR DESCRIPTION
**Pull request by aarti**
This causes errors in my testing, where I get an error that "Environment is not set" although I have set it.

Replacing the validate to use `validate_required_attributes` which is the idiomatic way in this project to handle attribute validation. 

```
GeoCLI.instance.env_name = "learning"
GeoCLI.instance.environment = GeoEngineer::Environment.new("learning")
project = GeoCLI.instance.environment.project("org","learning"){}
cert = project.resource("aws_acmpca_certificate_private_ca", "acbhq_dot_net") {...}
```

### CODE REVIEW

File: lib/geoengineer/project.rb [hunk 1]
```diff
@@ -12,16 +12,25 @@ class GeoEngineer::Project
...
 
   def initialize(org, name, environment, &block)
     @org = org
     @name = name
     @environment = environment
     instance_exec(self, &block) if block_given?
+    ## environments is an attribute that needs to be set, so lets make sure to set it.
+    if !@environment.nil? &&
+       !@environment.name
+      @_attributes[environments] = [@environment.name]
```
Comment (sds) [inline lib/geoengineer/project.rb]: Reading this looks a little odd—you're accessing a hash (`@_attributes`) using what is potentially an array (`environments`) as its key. Is that correct?

File: lib/geoengineer/project.rb [hunk 2]
```diff
@@ -12,16 +12,25 @@ class GeoEngineer::Project
...
   validate -> { all_resources.map(&:errors).flatten }
 
   def initialize(org, name, environment, &block)
     @org = org
     @name = name
     @environment = environment
     instance_exec(self, &block) if block_given?
+    ## environments is an attribute that needs to be set, so lets make sure to set it.
+    if !@environment.nil? &&
+       !@environment.name
```
Comment (sds) [inline lib/geoengineer/project.rb]: You can use the safe navigation operator here as well:

```ruby
if environment&.name
```

File: lib/geoengineer/project.rb (hunk 1 above)
Comment (aarti) [reply lib/geoengineer/project.rb]: That is correct. From inspecting the object in generated in tests "@_attributes={\"environments\"=>\"test\"}>}" we get this.

File: lib/geoengineer/project.rb (hunk 2 above)
Comment (aarti) [reply lib/geoengineer/project.rb]: Resolved

File: lib/geoengineer/project.rb (hunk 1 above)
Comment (aarti) [reply lib/geoengineer/project.rb]: hmm, may be not, in the tests it is a string.

Comment (sds) [reply lib/geoengineer/project.rb]: I might be misunderstanding you, but `@_attributes={"environments"=>"test"}>` means that you should write:
```ruby
@_attributes['environments']
```
...not:
```ruby
@_attributes[environments]
```
Does that clarify what I'm getting at?

Comment (aarti) [reply lib/geoengineer/project.rb]: yes

Comment (aarti) [reply lib/geoengineer/project.rb]: it was a typo

**Pull request merged**